includes/Article.php

```diff
@@ -1103,7 +1103,9 @@ public function showMissingArticle() {
 				$text = wfMsgNoTrans( 'noarticletext-nopermission' );
 			}
 		}
-		$text = "<div class='noarticletext'>\n$text\n</div>";
+
+		$text = "<div class='noarticletext'>$text</div>";
+		wfRunHooks( 'AddPortableInfoboxBuilderText', array( &$this, &$text, &$wgOut, &$errors ) );
```

Comment:
This hook should be called based on context, not our usage. My recommendation is something along the lines of "ShowMissingArticle::BeforeTextAppend"